### PR TITLE
Add unit tests for AgentRegistry page-context routing

### LINKED ISSUE #8: Add unit tests for Agent Registry
> _Originally created by @mingshl in [mingshl/opensearch-agent-server#5](https://github.com/mingshl/opensearch-agent-server/issues/5)_

## Description
Add unit tests for `src/orchestrator/registry.py` — the `AgentRegistry` class that maps page contexts to agent factories.

## Test cases
- [ ] Register an agent and retrieve it by page context
- [ ] Register multiple agents with different page contexts
- [ ] Duplicate page context registration raises error
- [ ] `list_agents()` returns all registered agents
- [ ] Fallback agent resolution when `is_fallback=True`
- [ ] No fallback agent raises appropriate error
- [ ] Agent with empty `page_contexts` list

## Location
`tests/unit/test_registry.py`

## References
- Source: `src/orchestrator/registry.py`

### PR DESCRIPTION
**Pull request by YiyuanMiao**
## Description

Adds unit tests for `AgentRegistry` (`src/orchestrator/registry.py`), which maps page contexts to the sub-agent that should handle a request. This is a tests-only change; no source code is modified.

### Coverage
- Register an agent and retrieve it by page context
- Register multiple agents with distinct page contexts
- Duplicate agent name raises `ValueError`
- Duplicate page context raises `ValueError`
- A conflicting registration does not leave the agent partially registered
- `list_agents()` returns all registered agents
- Default agent resolution via `is_default=True` / `get_default()`
- `get_default()` returns `None` when no default is registered
- An agent with an empty `page_contexts` list registers cleanly

### Notes for reviewers
- `get_default()` returns `None` when no default agent is registered; the fallback error is raised one layer up in `router.route()` (`src/orchestrator/router.py`). The tests assert `None` accordingly, matching the registry's actual responsibility.
- The original task description referenced an `is_fallback` field, but the actual API uses `is_default`. This was a naming mismatch in the task notes only — it does not affect the source code, and the tests are written against the real `is_default` field.

### Testing
- `pytest tests/unit/test_registry.py -v` → 11 passed
- `ruff check tests/unit/test_registry.py` → clean

### Check List
- [x] New functionality includes testing.
- [x] All tests pass.
- [ ] New functionality has been documented (N/A — tests only).
- [x] Commits are signed per the DCO using `--signoff`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
Closes #8

### CODE REVIEW

File: tests/unit/test_registry.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    registry.register(search)
+    registry.register(obs)
+
+    assert registry.get_agent_for_context("search-relevance") is search
+    assert registry.get_agent_for_context("observability") is obs
+    assert registry.get_agent_for_context("alerting") is obs
+
+
+# ---------------------------------------------------------------------------
```
Comment (mingshl) [inline tests/unit/test_registry.py]: nit: would you remove the extra comment line?

Comment (mingshl) [reply tests/unit/test_registry.py]: `# Register + retrieve by page context)` are enough
  without the redundant # ---- lines below them. Keeps it cleaner.

File: tests/unit/test_registry.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    reg = _registration("search-agent", page_contexts=["search-relevance"])
+
+    registry.register(reg)
+
+    resolved = registry.get_agent_for_context("search-relevance")
+    assert resolved is reg
+    assert resolved.name == "search-agent"
+
+
+def test_get_agent_for_unknown_context_returns_none() -> None:
```
Comment (mingshl) [inline tests/unit/test_registry.py:63]: maybe add a test when  get_agent_for_context on unknown context when registry is empty

**Comment by mingshl** [comment]
Thanks for your contribution! left some comments but overall the unit tests look good, we also need some integration test coverage if you are shooting for the next task.

**Comment by YiyuanMiao** [comment]
Added a test for get_agent_for_context on an empty registry, and removed the redundant separator lines. Thanks!

File: tests/unit/test_registry.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    registry.register(_registration("agent-a", page_contexts=["shared-ctx"]))
+
+    with pytest.raises(ValueError):
+        registry.register(
+            _registration("agent-b", page_contexts=["fresh-ctx", "shared-ctx"])
+        )
+
+    # agent-b must not be in the registry, and its first (non-conflicting)
+    # context must not have leaked into the map before the conflict was hit.
+    assert all(a.name != "agent-b" for a in registry.list_agents())
```
Comment (wrigleyDan) [inline tests/unit/test_registry.py:114]: Should we also verify that "fresh-ctx" (the non-conflicting context registered before the conflict was hit) wasn't written into the page-context map?

```suggestion
    assert all(a.name != "agent-b" for a in registry.list_agents())
    # verify that "fresh-ctx" (the non-conflicting context registered before the conflict was hit) wasn't
    # written into the page-context 
    assert registry.get_agent_for_context("fresh-ctx") is None
```

**Review by wrigleyDan — CHANGES_REQUESTED** [review]

**Comment by mingshl** [comment]
@YiyuanMiao are you currently working on this PR?

**Comment by jiapingzeng** [comment]
Hi @YiyuanMiao, could you please rebase against main and fix the lint/format check? Otherwise this PR looks good. Thanks!
